imap_processing/spice/geometry.py

```
@@ -388,8 +393,21 @@ def get_rotation_matrix(
         `et` is a np.ndarray, the returned rotation matrix is of shape `(n, 3, 3)`
         where `n` matches the number of elements in et.
     """
+
+    def pxform_error_handler(  # type: ignore[no-untyped-def]
+        *arg, **kwargs
+    ):  # numpydoc ignore=GL08
+        try:
+            return spiceypy.pxform(*arg, **kwargs)
+        except spiceypy.utils.exceptions.SpiceNOFRAMECONNECT:
```

This implementation concerns me. I want this function to fail loudly by default rather than just logging an exception level message. I would be more comfortable with adding a keyword argument called something like `allow_spice_noframeconnect` that defaults to False.
```
try:
    return spiceypy.pxform(*arg, **kwargs)
except spiceypy.utils.exceptions.SpiceNOFRAMECONNECT as e:
    if not allow_spice_noframeconnect:
        raise(e)
    logger.debug(f"Returning NaN rotation matrix due to spiceypy error: {e}")
    return np.full((3, 3), np.nan)
```

---

Makes sense, done!